src/c++/perf_analyzer/docs/llm.md

```diff
@@ -0,0 +1,136 @@
+# Benchmarking LLM
+
+### Setup model/server environment
+
+Follow step 1 from the [Triton + vLLM tutorial](https://github.com/triton-inference-server/tutorials/blob/main/Quick_Deploy/vLLM/README.md).
+
+### Prefill
+
+In this benchmarking scenario, we want to measure the effect of input prompt size on first-token latency. We issue single request to the server of fixed input sizes and request the model to compute at most one new token. This essentially means one pass through the model.
+
+#### 1. Run these commands
+
+```bash
+PATH_TO_MODEL_PY="model_repository/vllm/1/model.py"
+MAX_TOKENS=1
+sed -i "128s/.*/\ \ \ \ \ \ \ \ params_dict[\"max_tokens\"] = ${MAX_TOKENS}/" ${PATH_TO_MODEL_PY}
```

Comment:
I think this is a bit hacky and the tutorial might break when the vLLM example changes. @tanmayv25 do you think we can update the example to use input tensors instead of parameters? This would provide a path for profiling these models in PA as well.

Reply:
Yeah we were left with few options. Either rip apart PA to allow passing down the max tokens/ignore eos as request parameters or hard-code it by modifying the model here in the guide.

Reply:
Sending the SampleParameters by request parameters was an intentional decision. This would most like remain as such for vLLM usability.

Reply:
> Sending the SampleParameters by request parameters was an intentional decision. This would most like remain as such for vLLM usability.

@tanmayv25 do you mind elaborating more on why request parameters were chosen over inputs in the vLLM tutorial?

Reply:
The abstraction helps separating [SamplingParams](https://github.com/vllm-project/vllm/blob/32b6816e556f69f1672085a6267e8516bcb8e622/vllm/sampling_params.py#L7) from prompt. 
It reduces the number of input tensors in the config.pbtxt and client to have individually specify them as separate tensor.

Reply:
I think we will start off with the initial phase of this LLM benchmark requiring the user modify their model to accept `MAX_TOKENS` and `IGNORE_EOS` inputs to affect model behavior to enable people to run these benchmarks earlier than later.

Then, later, since PA development is slow, we can implement custom parameters support in PA.

Reply:
@Tabrizian @tanmayv25 friendly ping

Reply:
@matthewkotila  It would be good to mention this hack in these docs. We should call out that because PA currently do not support specifying parameters with its requests we are modifying the model.py. Which would not be needed once PA starts supporting request parameters.

Reply:
I'm still not convinced that using parameters provides value over inputs but if we have to stick with that the only paths forward would be for PA to support parameters OR hardcoding the values as you did here.

Reply:
> I'm still not convinced that using parameters provides value over inputs
It is just a more pleasant way of providing scalars to the model. Instead of creating numpy arrays for each of the arguments of SampleParameters and passing them to Triton client, we are directly sending these scalars as request parameter. There is nothing wrong nor stopping us from specifying them as separate input tensors. But specifying these attributes as request parameters considerably reduce the boilerplate code on the client side. Additionally, the SamplingParams abstraction is easily understood by vLLM users. 
I think this was one of the feedback received previously. We can also improve our python client library to reduce the boilerplate code by accepting scalars as input.